modules/layers/src/scatterplot-layer/scatterplot-layer-fragment.glsl.ts

```diff
@@ -61,7 +61,13 @@ void main(void) {
   } else if (filled) {
     gl_FragColor = vFillColor;
   } else {
-    discard;
+    #ifdef XCLIPSE920
```

Comment:
Are there other shaders on which this fails? E.g. IconLayer also performs a discard

Reply:
I went thru most of the samples and have only seen these including Scatterplot layer failing on S22. I believe it's the only shader failing on Xclipse 920.
I suspect their shader compiler might miscompile discard in a specific context. Unfortunately I don't have mode precise findings.

Reply:
> I suspect their shader compiler might miscompile discard in a specific context.

Yes.. if that is the case it might even be possible to just rewrite the logic, keep the functionality intact and avoid triggering the bug. It might be worth testing a few variations to see if we can pinpoint what triggers it.

The shader compilers in the drivers can be finicky. I recall a case where basically just one line change in the PathLayer shader caused the compilation time for that shader on Windows systems to suddenly jump to 30s.

Regardless, I think this fix as it stands is still worthwhile if it fixes this long-standing issue.

Reply:
That could be true, but as I fiddled with it that led to duplicate conditions and certainly wasn't as optimal at the original implementation. This proposed fix has some advantages:
- it is explicit, and makes one aware of the issue,
- only affects minor scenario of unfilled markers in terms of performance and occlusion robustness.

Reply:
Somewhat less obtrusive fix identified, please see #8053.